src/Compilation.zig

```diff
@@ -379,23 +393,35 @@ pub const AllErrors = struct {
             indent: usize,
         ) anyerror!void {
             const stderr = stderr_file.writer();
+            // The length of the part before the error message; e.g. "file.zig:4:5: error: ".
+            var prefix_len: usize = 0;
             switch (msg) {
                 .src => |src| {
                     try stderr.writeByteNTimes(' ', indent);
+                    prefix_len += indent;
                     ttyconf.setColor(stderr, .Bold);
                     try stderr.print("{s}:{d}:{d}: ", .{
                         src.src_path,
                         src.line + 1,
                         src.column + 1,
                     });
+                    prefix_len += std.fmt.count("{s}:{d}:{d}: ", .{
```

Comment:
It's unfortunate that `stderr.print` etc. don't return the written bytes... has this been discussed before? I understand that it might be too annoying to have `_ = ` before every line but at the same time there are definitely use cases for the byte count like you can see here.

Reply:
You can use an `std.io.countingWriter` for that.

Reply:
You're right! Thank you!

Reply:
Just write the escape codes to the inner writer.

Reply:
Ah, you guys are so right!